dhcp-relay/dhcp_kern_xdp.c

```diff
@@ -438,57 +631,50 @@ int xdp_dhcp_relay(struct xdp_md *ctx) {
 
 			if (write_dhcp_option_255(ctx, option_offset)) {
 				bpf_printk("Could not write DHCP option 255 at offset %i", option_offset);
-				return XDP_ABORTED;
+				//return XDP_ABORTED;
+				break;
 			}
 
 			bpf_printk("Wrote DHCP option 255 at offset %i, returning XDP_PASS", option_offset);
 
 			break;
 		}
+
 		pos++;
 
+		/* Verifier check */
+		if (pos + 1 > data_end) {
+			break;
+		}
+
 		option_length = *pos;
 		option_offset += option_length + 2;
 
+		/* Verifier check */
```

Comment:
These repeated 'verifier check' comments are just noise. There are also way too many checks one after another at the end here, some of them could definitely be collapsed. This may be easier to figure out after some of the other refactoring mentioned above.